lib/internal/url.js

```diff
@@ -1265,6 +1265,10 @@ function domainToUnicode(domain) {
   return _domainToUnicode(`${domain}`);
 }
 
+function decodeAuth(str) {
+  return decodeURIComponent(str).replace(':', '%3A').replace('/', '%2F');
```

Comment:
it would be unexpected outcome with undefined `url.password` or `url.username`

Reply:
Is it possible for `url.username` to be `undefined`? Isn't it an empty string instead?